File: roundRobin/Processo.java
package roundRobin;

import java.util.LinkedList;
import java.util.NoSuchElementException;

public class Processo extends Object {
	
	private String nome;
	private int momento;
	private int quantun;
	private LinkedList<String> estados;
	
	public Processo(String nome, int momento, int quantun) {
		super();
		this.nome = nome;
		this.momento = momento;
		this.quantun = quantun;
		estados = new LinkedList<String>();
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public int getMomento() {
		return momento;
	}

	public void setMomento(int momento) {
		this.momento = momento;
	}

	public int getQuantun() {
		return quantun;
	}

	public void setQuantun(int quantun) {
		this.quantun = quantun;
	}

	public String[] getEstados() {
		return estados.toArray(new String[1]);
	}

	public void addEstado(String estado) {
		this.estados.add(estado);
	}
	
	public boolean isOver() {
		try {
			if(estados.getLast()=="f") {
				return true;
			}
		} catch (NoSuchElementException e) {
			return false;
		}
		return false;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((nome == null) ? 0 : nome.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Processo other = (Processo) obj;
		if (nome == null) {
			if (other.nome != null)
				return false;
		} else if (!nome.equals(other.nome))
			return false;
		return true;
	}
	
	public String toString() {
		return nome+"|"+momento+"|"+quantun;
	}

}

File: Interativo/Processo.py
class Processo(object):

	def __init__(self, nome, prioridade=0):
		self.nome = nome
		self.estados = []
		self.prioridade = prioridade
		self.prioridadeDef = prioridade

	def addEstado(self, estado):
		self.estados.append(estado)

	def getEstados(self):
		retorno = ''
		for i in self.estados:
			retorno+=i
		return retorno

	def prioUp(self,quant):
		self.prioridade+=quant

	def prioRedef(self):
		self.prioridade = self.prioridadeDef

	def __eq__(self, obj):

		if obj == None:
			return False
		if self.__class__ != obj.__class__:
			return False
		if self.nome == obj.nome:
			return True
		return False

	def __str__(self):
		retorno = 'Status '+self.nome+': '
		try:
			retorno+=self.estados[-1]
		except IndexError:
			retorno+='None'
		return retorno

	def isBloked(self):
		try:
			if self.estados[-1] == 'b':
				return True
		except IndexError:
			return False
		return False
 